@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -625,15 +760,22 @@ export class VisibleTileSet {
                         searchDirection === SearchDirection.BOTH ||
                         searchDirection === SearchDirection.DOWN
                     ) {
-                        const tileKey = TileKey.fromMortonCode(tileKeyCode);
+                        const { offset, mortonCode } = extractOffsetAndMortonKeyFromKey(
+                            tileKeyCode
+                        );
+                        const tileKey = TileKey.fromMortonCode(mortonCode);
                         tilingScheme.getSubTileKeys(tileKey).forEach(childTileKey => {
-                            const childTileCode = childTileKey.mortonCode();
-                            const childTile = tileCache.get(childTileCode);
-
+                            const childTileCode = getKeyForTileKeyAndOffset(childTileKey, offset);
                             checkedTiles.add(childTileCode);
-                            if (childTile !== undefined && childTile.hasGeometry) {
+
+                            const childTileOrProxy = tileCache.get(childTileCode);
+                            if (
+                                childTileOrProxy !== undefined &&
+                                childTileOrProxy.tile.hasGeometry &&
+                                !childTileOrProxy.tile.disposed
```

Comment:
> !childTileOrProxy.tile.disposed

This!
Why not allowing clone - which should be independent from original tile - to live if parent has been disposed?

Another pros for reference-counting underlying buffers & geometries.

Reply:
> Why not allowing clone - which should be independent from original tile - to live if parent has been disposed?

Technically the clone is still alive if the parent has been disposed, and that is the problem. The TileProxy and the Tile are two separate entries in the cache. It is possible that the Tile has been evicted but not the TileProxy. In that case, if we don't make a check for it, we will try to render something which Proxies nothing, and therefore we get a blank area (this was visible when zooming in / out in the offsetted region).

The problem is that the Tile has no reference to the TileProxy, so when the Tile is evicted, it isn't easy to inspect the cache and remove all associated TileProxys, but I don't think this is a major, they will eventually be removed anyway.